spec/javascripts/unit/admin/products/services/unit_prices_spec.js.coffee

```diff
@@ -0,0 +1,140 @@
+      
+      expect(UnitPrices.unit(scale, unit_type)).toEqual "kg"
+    it "with price: 2, scale: 1000 and unit_value: 500", ->
+      price = 2
+      scale = 1000
+      unit_value = 500
+      expect(UnitPrices.price(price, scale, unit_type, unit_value)).toEqual 4
+      
+      expect(UnitPrices.unit(scale, unit_type)).toEqual "kg"
+
+
+  describe "get correct unit price duo unit/value for volume", ->
+    unit_type = "volume"
+    it "with scale: 1", ->
+      price = 1
+      scale = 1
+      unit_value = 1
+      expect(UnitPrices.price(price, scale, unit_type, unit_value)).toEqual 1
+      
+      expect(UnitPrices.unit(scale, unit_type)).toEqual "L"
+    it "with price: 2 and unit_value: 0.5", ->
+      price = 2
+      scale = 1
+      unit_value = 0.5
+      expect(UnitPrices.price(price, scale, unit_type, unit_value)).toEqual 4
+      
+      expect(UnitPrices.unit(scale, unit_type)).toEqual "L"
+    it "with price: 2, scale: 0.001 and unit_value: 0.01", ->
+      price = 2
+      scale = 0.001
+      unit_value = 0.01 
+      expect(UnitPrices.price(price, scale, unit_type, unit_value)).toEqual 200
+      
+      expect(UnitPrices.unit(scale, unit_type)).toEqual "L"
+    it "with price: 20000, scale: 1000 and unit_value: 10000", ->
+      price = 20000
+      scale = 1000
+      unit_value = 10000 
+      expect(UnitPrices.price(price, scale, unit_type, unit_value)).toEqual 2
+      
+      expect(UnitPrices.unit(scale, unit_type)).toEqual "L"
+
+  describe "get correct unit price duo unit/value for items", ->
+    unit_type = "items"
+    scale = null
+    it "with price: 1 and unit_value: 1", ->
+      price = 1
+      unit_value = 1
+      expect(UnitPrices.price(price, scale, unit_type, unit_value)).toEqual 1
+      
+      expect(UnitPrices.unit(scale, unit_type)).toEqual "item"
+    it "with price: 1 and unit_value: 10", ->
+      price = 1
+      unit_value = 10
+      expect(UnitPrices.price(price, scale, unit_type, unit_value)).toEqual 0.1
+      
+      expect(UnitPrices.unit(scale, unit_type)).toEqual "item"
+    it "with price: 10 and unit_value: 1", ->
+      price = 10
+      unit_value = 1
+      expect(UnitPrices.price(price, scale, unit_type, unit_value)).toEqual 10
+      
+      expect(UnitPrices.unit(scale, unit_type)).toEqual "item"
+
+
+  describe "get correct unit price duo unit/value for weight in imperial system", ->
+    unit_type = "weight"
+    it "with price: 1 and scale/unit_value: 28.35 (OZ)", ->
+      price = 1
+      scale = 28.35
+      unit_value = 28.35
+      expect(UnitPrices.price(price, scale, unit_type, unit_value)).toEqual 16
+      
+      expect(UnitPrices.unit(scale, unit_type)).toEqual "lb"
+    it "with price: 1 and scale/unit_value: 453.6 (LB)", ->
+      price = 1
+      scale = 453.6
+      unit_value = 453.6
+      expect(UnitPrices.price(price, scale, unit_type, unit_value)).toEqual 1
+      
+      expect(UnitPrices.unit(scale, unit_type)).toEqual "lb"
```

Comment:
I think we misunderstood here. What I meant is that each `it` block should be separate from the other. This way readers can distinguish each test. Now, it looks like the last expectation in a test belongs to the next one I find it very weird and it difficult my reading, but it's just my opinion.

Reply:
I agree with you.
So, you preferrer: 

```coffeescript
  describe "get correct unit price duo unit/value for weight in imperial system", ->
    unit_type = "weight"

    it "with price: 1 and scale/unit_value: 28.35 (OZ)", ->
      price = 1
      scale = 28.35
      unit_value = 28.35
      expect(UnitPrices.price(price, scale, unit_type, unit_value)).toEqual 16
      expect(UnitPrices.unit(scale, unit_type)).toEqual "lb"

    it "with price: 1 and scale/unit_value: 453.6 (LB)", ->
      price = 1
      scale = 453.6
      unit_value = 453.6
      expect(UnitPrices.price(price, scale, unit_type, unit_value)).toEqual 1
      expect(UnitPrices.unit(scale, unit_type)).toEqual "lb"
```
To me, it's also better.

Reply:
yes, exactly.